draft/0001-organization.md

```diff
@@ -0,0 +1,273 @@
+
+It is RECOMMENDED to only use lowercase letters with specific symbols (`:_`) for permissions.
+
+### 4.4. Organization role
+
+An organization role has a unique identifier and contains a set of organization permissions. Each organization role in the template SHOULD only be granted with predefined permissions in the same template. An organization role MAY have no permission granted.
+
+### 4.5. Assign roles to organization members
+
+Before assigning roles to an identity, the identity MUST have a membership with the organization, making them an organization member. An organization member can be assigned multiple roles within the organization, and each role held by the member must be predefined in the organization template. The complete set of permissions granted to an organization member is the union of all permissions granted by their roles.
+
+### 4.6. Role-based access control
+
+To authorize an actor (identity) to perform a specific action within an organization context, the actor MUST be assigned at least one role that contains the desired permission (scope).
+
+If the actor does not specify the organization context, the authorization MUST fail.
+
+It's important to note that organization roles are not considered in authorization. The actor is authorized to perform the action based solely on the organization permission, regardless of their organization role.
+
+### 4.7. Relationship to existing specifications
+
+Organization permissions (scopes) are distinct from the `scope` in OpenID Connect and [Resource Indicators](https://www.rfc-editor.org/rfc/rfc8707.html#name-access-token-request). While these existing specifications share terminology like "scope" and "role" with organization, they are designed to address different authentication and authorization challenges. Organization scopes and roles should be treated separately.
+
+## 5. Authentication
+
+In OpenID Connect, the [ID Token](https://openid.net/specs/openid-connect-core-1_0.html#IDToken) is a JWT token containing user identity information. In addition to the [Standard Claims](https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims), we introduce new claims to the ID token to support organization.
+
+### 5.1 Organization claims
+
+| Name               | Type                  | Description                                                                                                                                                                                          |
+| ------------------ | --------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| organization_ids   | array of strings      | The unique organization identifiers that the user has a membership with.                                                                                                                             |
```

Comment:
Can we use `org_ids` to reduce the JWT length, the same as other claims like "iss" and "exp".

Reply:
i don't like three-letter abbreviations, but i'm good with it. we can vote for this

Reply:
use org